tools/database/src/analyse.rs

```diff
@@ -0,0 +1,271 @@
+            let column_number_of_pairs = local_key_sizes.values().sum::<usize>();
+            let column_size =
+                local_key_sizes.iter().map(|(&size, &count)| size * count).sum::<usize>();
+            let column_family = ColumnFamilyCountAndSize {
+                number_of_pairs: column_number_of_pairs,
+                size: column_size,
+            };
+            guard.insert(col_family.to_string(), column_family);
+        }
+
+        update_map(&key_sizes, &local_key_sizes);
+        update_map(&value_sizes, &local_value_sizes);
+    });
+
+    let key_sizes: Vec<(usize, usize)> = key_sizes.lock().unwrap().clone().into_iter().collect();
+    let value_sizes: Vec<(usize, usize)> =
+        value_sizes.lock().unwrap().clone().into_iter().collect();
+    let column_families: Vec<(String, ColumnFamilyCountAndSize)> =
+        column_families_data.lock().unwrap().clone().into_iter().collect();
+
+    DataSizeDistributionResults::new(key_sizes, value_sizes, column_families)
+}
+
+fn get_column_family_options(
+    input_col: &Option<String>,
+) -> (Vec<String>, Vec<ColumnFamilyDescriptor>) {
+    match input_col {
+        Some(col_name) => {
+            let mut opts = Options::default();
+            let maybe_db_col = get_db_col(&col_name);
+            if let Some(db_col) = maybe_db_col {
+                if db_col.is_rc() {
+                    opts.set_merge_operator(
+                        "refcount merge",
+                        RocksDB::refcount_merge,
+                        RocksDB::refcount_merge,
+                    );
+                    opts.set_compaction_filter(
+                        "empty value filter",
+                        RocksDB::empty_value_compaction_filter,
+                    );
+                }
+            }
+            (vec![col_name.clone()], vec![ColumnFamilyDescriptor::new(col_name, opts)])
+        }
+        None => {
+            let col_families = get_all_col_family_names();
+            (
+                col_families.iter().map(|db_col| col_name(*db_col).to_string()).collect(),
+                col_families
+                    .iter()
+                    .map(|db_col| {
+                        ColumnFamilyDescriptor::new(
+                            col_name(*db_col),
+                            rocksdb_column_options(*db_col),
+                        )
+                    })
+                    .collect(),
+            )
+        }
+    }
+}
+
+impl AnalyseDatabaseCommand {
+    pub fn run(&self, home: &PathBuf) -> anyhow::Result<()> {
+        // Set db options for maximum read performance
+        let mut opts = Options::default();
+        opts.set_max_open_files(20_000);
+        opts.increase_parallelism(std::cmp::max(1, num_cpus::get() as i32 / 2));
+
+        // Define column families
+        let (col_families, col_families_cf) = get_column_family_options(&self.column);
+
+        let db = DB::open_cf_descriptors_read_only(
```

Comment:
You can you `iter_raw_bytes` to skip refcount logic.

Generally I agree that having a separate tool with minimal dependencies is good. But I don't think you actually achieve your tool being independent from our codebase since you still take a bunch of internals (private functions) from `rocksdb.rs` to properly configure this. It means that now any changes to our rocksdb config (such as adding another column with merge operator) would need to be applied to this tool to keep it working which makes it fragile and introduce additional code maintenance overhead.

Reply:
Aha true I will try `raw_iter_bytes`, I think that should work.

I see your point, it makes sense in this context